### DISCUSSION TITLE
Is it possible to have option like `--update-env` that will pull updated environmental variables after elevated command finishes and apply this env to non-elevated shell?

### DISCUSSION THREAD

**Discussion by soredake** [gerardog/gsudo#328] [Q&A]
For example, I want to install Node.js using sudo winget install -h nodejs, i have WingetPathUpdater installed, the problem is, it will update env only in elevated scope, so if I want to execute node or npm from non-elevated scope I need to will need to manually refresh env. Is it possible to have an option like `--update-env` that will pull updated environmental variables after the elevated command finishes and apply this env to the non-elevated shell?

https://github.com/jazzdelightsme/WingetPathUpdater/issues/4

**Comment by gerardog** [comment]
Hi soredake,

I understand the convenience such a feature would offer, especially in scenarios like the one you've described with Node.js installation.

However, an .EXE app can't update the invoker (i.e. non-elevated) shell's environment variables. It could be done by a .cmd batch file such as chocolatey's built-in refreshenv.cmd. Adding this to this project extends beyond the scope of what `gsudo` is designed for. `gsudo` primary purpose is to provide privilege elevation, not environment management.

That said, I can recommend an alternative approach. You might want to look into [RefrEnv](https://github.com/badrelmers/RefrEnv), a tool specifically designed for refreshing environment variables in Windows. It can help update your non-elevated shell's environment variables more easily, which seems to align with your requirements.

Again, I appreciate your input and hope this alternative can be of help!
